src/app/chat/views/channel.nim

```diff
@@ -296,11 +295,9 @@ QtObject:
     result = selectedChannel.muted  
 
   proc getChatItemById*(self: ChannelView, id: string): QObject {.slot.} =
-    if self.chatItemViews.hasKey(id): return self.chatItemViews[id]
     let chat = self.getChannelById(id)
     let chatItemView = newChatItemView(self.status)
     chatItemView.setChatItem(chat)
-    self.chatItemViews[id] = chatItemView
```

Comment:
@pinebit you're probably aware of this, still pointing it out to be sure:

`getChatItemById` returns a `QObject` which, if not kept in memory, can potentially be garbage collected while the view still references it (that's why I've introduced this table originally).

Has this somehow changed?

Reply:
I checked the usage of this function. It is only used in two places when we display the corresponding context menu. The reference to this object is kept by `ChatContextMenu` and in turn by `CreateChannelPopup`. It is Okay if it will be destroyed if both the menu and the popup are closed.
The table you introduced added another side effect. It cached old chat data, therefore after editing a chat title and/or description, it displayed old data when the user entered the popup second time.

Reply:
Gotcha, thanks for clarifying!